realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -346,14 +310,58 @@ protected void onError(LogoutResponse response) {
     }
 
     /**
+     * Change this users password. This is done synchronously and involves the network, so calling this method on the
+     * Android UI thread will always crash.
+     *
+     * @param newPassword the users new password.
+     * @throws IllegalArgumentException if not on a Looper thread.
```

Comment:
This restriction is strange, I don't see why user cannot call this method on a non-UI looper thread. Also, calling this on UI thread will surely block the UI thread, but it is user's responsibility to use the async version in this case I think.